testsuite/smoke-test/src/aptos_cli/validator.rs

```diff
@@ -69,6 +75,346 @@ async fn test_show_validator_set() {
+            genesis_config.allow_new_validators = true;
+            genesis_config.epoch_duration_secs = 4;
+            genesis_config.recurring_lockup_duration_secs = 4;
+            genesis_config.voting_duration_secs = 3;
+            genesis_config.rewards_apy_percentage = 10;
+        }))
+        .build_with_cli(0)
+        .await;
+
+    let transaction_factory = swarm.chain_info().transaction_factory();
+
+    let mut validators: Vec<_> = swarm.validators().collect();
+    validators.sort_by_key(|v| v.name());
+
+    let validator_cli_indices = validators
+        .iter()
+        .map(|validator| {
+            cli.add_account_to_cli(
+                validator
+                    .account_private_key()
+                    .as_ref()
+                    .unwrap()
+                    .private_key(),
+            )
+        })
+        .collect::<Vec<_>>();
+    let rest_clients = validators
+        .iter()
+        .map(|validator| validator.rest_client())
+        .collect::<Vec<_>>();
+    let addresses = validators
+        .iter()
+        .map(|validator| validator.peer_id())
+        .collect::<Vec<_>>();
+
+    println!("{:?}", addresses.iter().enumerate().collect::<Vec<_>>());
+
+    async fn get_state_and_validator_set(rest_client: &Client) -> (State, HashMap<PeerId, u64>) {
+        let response = rest_client
+            .get_resource(CORE_CODE_ADDRESS, "0x1::stake::ValidatorSet")
+            .await
+            .unwrap();
+        let (result, state) = response.into_parts();
+        let validator_set = to_validator_set(&result);
+        let validator_to_voting_power = validator_set
+            .active_validators
+            .iter()
+            .chain(validator_set.pending_inactive.iter())
+            .map(|v| (v.account_address, v.consensus_voting_power()))
+            .collect::<HashMap<_, _>>();
+        (state, validator_to_voting_power)
+    }
+
+    println!(
+        "{:?}",
+        get_state_and_validator_set(&rest_clients[0]).await.1
+    );
+
+    rest_clients[2]
+        .set_failpoint(
+            "consensus::send::broadcast_proposal".to_string(),
+            "100%return".to_string(),
+        )
+        .await
+        .unwrap();
+
+    reconfig(
+        &rest_clients[0],
+        &transaction_factory,
+        swarm.chain_info().root_account(),
+    )
+    .await;
+
+    let (start_2_failures_state, start_2_failures_validator_set) =
+        get_state_and_validator_set(&rest_clients[0]).await;
+    println!(
+        "Node 2 ({}) starts failing: at epoch {} and version {}, set: {:?}",
+        addresses[2],
+        start_2_failures_state.epoch,
+        start_2_failures_state.version,
+        start_2_failures_validator_set
+    );
+
+    tokio::time::sleep(Duration::from_secs(5)).await;
+
+    reconfig(
+        &rest_clients[0],
+        &transaction_factory,
+        swarm.chain_info().root_account(),
+    )
+    .await;
+    println!("#0.5");
```

Comment:
These println!s look like they're for debugging. Can we remove them before merging the PR?

Reply:
:( forgot about those

Reply:
But I'll remove the number ones. The rest are useful for root-causing if test breaks.